OADP-2130: rework of the introduction section (#62396)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/oadp-intro.adoc

:_content-type: ASSEMBLY
[id="oadp-introduction"]

> = Introduction to OADP

Reviewer: ```suggestion
= Introduction to {oadp-full}
```

> [id="oadp-introduction"]
> = Introduction to {oadp-full}
> include::_attributes/common-attributes.adoc[]

Reviewer 2: In order to have the `{oadp-full}` variable render in the heading on the Customer Portal, you need to move the `include::` statement above the heading. You can see an example in the [contributor guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata).

This quirk of the Customer Portal is something we only found out about recently. It was brought up at the OpenShift docs weekly team meeting today, 21 August 2023.
```suggestion
[id="oadp-introduction"]
include::_attributes/common-attributes.adoc[]
= Introduction to {oadp-full}
```

:context: oadp-api
:namespace: openshift-adp
:local-product: OADP

toc::[]

> The {oadp-first} product is intended to safeguard customer applications on the OpenShift Platform. OADP provides disaster recovery protection for OpenShift applications, application related cluster resources, persistent volumes, and internal images.

Reviewer: ```suggestion
The {oadp-first} product safeguards customer applications on the {product-title}. OADP provides disaster recovery protection for {product-title} applications, application-related cluster resources, persistent volumes, and internal images.
```

> OADP can be used to back up containerized applications and virtual machines (VMs).

Reviewer 3: might be confusing
vm is also the short-hand name of the VirtualMachine Resouce type 
When br VM, it is not only the vm resource

Author: but it says *virtual machine* in front of it so there is no confusion

> OADP is not intended to be a disaster recovery solution for *etcd* or OpenShift Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].

Reviewer: ```suggestion
OADP is not a disaster recovery solution for *etcd* or {product-title} Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```
OR
```suggestion
OADP cannot be a disaster recovery solution for *etcd* or OpenShift Operators. For more information on etcd, see xref:../../backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.adoc#backup-etcd[Backing up etcd].
```

> There exists a multitude of approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.

Reviewer: I am not aware of the technicalities. But are those approaches = the following list of APIs?
If so, my suggestion:
```suggestion
OAP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.
```
else
 you might have to explain those multiple approaches.

Author: (giggle) @snarayan-redhat - you spelt OADP wrong! What a bad reviewer! LOL

Seriously, thanks for the great help you gave with your review.

> OADP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.

Reviewer 2: +1 to Steven's comment. I am not sure that adding links will clarify the sentence. I find the infinitive phrases difficult to follow. Is this sentence is describing one, very specific action that these APIs enable, or if it is a list of capabilities that these APIs share? Do the APIs use namespaces and labels to avoid backing up the wrong resources? 

I understand that some of this content is a placeholder for future revisions.

Author: @michaelryanpeter - I am sorry not repeating myself

Reviewer 4: @anarnold97, I agree with Michael and Steven. I think you should comment out lines 17-26 until you can link to the API information. In this state, it's not providing additional value.

Author: I originally had it separated, but the first reviewer wanted a change and suggested adding that sentence. So can I go back to something like the original:
```
There exists numerous approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels. 

OADP provides the following APIs:

    Backup

    Restore

    Schedule

    BackupStorageLocation

    VolumeSnapshotLocation
```
These would then link to the new main sections in the document.

> OAP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.

Author: ```suggestion
OADP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.
```

> OADP provides the following APIs to enable multiple approaches to customize a backup to avoid backing up inappropriate resources by using namespaces or labels.

Author: Please remember this is part of a re-work and once done, I need to come back and add links, which will hopefully clarify

> OADP provides the following APIs:

Reviewer 3: there are more velero resource types 
but I assume we include only the customer-facing ones, correct?

backuprepositories
backups
backupstoragelocations
deletebackuprequests
downloadrequests
podvolumebackups
podvolumerestores
restores
schedules
serverstatusrequests
volumesnapshotlocations

Author: Yes

* Backup
* Restore
* Schedule
* BackupStorageLocation
* VolumeSnapshotLocation

> // once finished re-work come back and add doc links to the APIs

Reviewer 3: was this missed?

Author: No. It is a note for myself

Author: Please remember, that this restructure is split into three parts. Once each is changed, I need to come back and add links to these sections. 

However, until I have the sections approved I cannot link cleanly.

Reviewer 3: @anarnold97 
This is still WIP, correct or is it ready?

Author: It is mainly finished - I was trying to get the restructure done as there are so many links. Then do the re-write... but it looks like QE has decided I am doing everything at once. LOL

> * xref:../../backup_and_restore/application_backup_and_restore/oadp-api.adoc#oadp-api[Schedule]

Reviewer 5: this link should redirect to the Scheduling Backups section.
https://62396--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications#oadp-scheduling-backups_backing-up-applications

I dont see anything related to Schedule on the current link it redirects to.

> * VolumeSnapshotLocation

Reviewer 6: @anarnold97 I am assuming here that the link for  `BackupSnapshotLocation` is the same as for `VolumeSnapshotLocation`. 
I am not sure which option would be less confusing, leaving `VolumeSnapshotLocation` without a link, or provide the same link for both. I defer to your guidance.

Author: Thanks @mburke5678 - will make this change later this evening
